c10/util/order_preserving_flat_hash_map.h

```diff
@@ -660,14 +653,18 @@ class sherwood_v3_table : private EntryAlloc, private Hasher, private Equal
         int8_t old_max_lookups = max_lookups;
         max_lookups = new_max_lookups;
         num_elements = 0;
-        for (EntryPointer it = new_buckets, end = it + static_cast<ptrdiff_t>(num_buckets + old_max_lookups); it != end; ++it)
-        {
-            if (it->has_value())
-            {
-                emplace(std::move(it->value));
-                it->destroy_value();
-            }
+
+        auto start = sentinel->next;
```

Comment:
Note: a reasonable invariant to maintain would be that the actual hash table layout of the table would look exactly the same as flat_hash_map and the only change would be the linked list for user traversals. But using the linked list order here is going to change the way the resized hash table appears. I do not know if that leads to any perf issues. One concern here is that what used to be a linear scan through a pretty full hash table is now a slow linked list traversal.

Reply:
Yes, this likely decreases perf of the hash map. Why do we need to change this? Can't we keep it as a linear scan? Did you do benchmarks?

There's also many other places in the code that use iterators to iterate through all elements, all of these would be slower now. We should avoid that and allow them to ignore order when iterating.

Reply:
As far as i can tell even if I replace this with a linear scan, I will need to iterate through the linked list to preserve insertion order afterwards. 

Unless i'm missing a solution here ?

Reply:
There are two issues here: (1) a linked list traversal is potentially slower (probably not fixable, after all we need to maintain the linked list). (2) The linked list traversal is a _different_ order from what was before, but re-hashing is sensitive to to the order things are inserted. I don't know enough about this hash table implementation to say whether it matters